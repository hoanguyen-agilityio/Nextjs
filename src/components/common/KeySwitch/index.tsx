// Third party
import { Switch, SwitchProps } from '@nextui-org/react';

interface IKeySwitch extends SwitchProps {
  title: string;
  describe: string;
  classWrapper?: string;
  ariaLabel: string;
}

const KeySwitch = ({
  title,
  describe,
  classWrapper,
  ariaLabel,
  ...switchProps
}: IKeySwitch) => {
  return (
    <div className={`flex justify-between ${classWrapper}`}>
      <div>
        <span className="text-lg text-black-900 dark:text-white">{title}</span>
        <p className="font-normal text-xsm text-ratio-200 dark:text-grayBlue-400">
          {describe}
        </p>
      </div>
      <Switch {...switchProps} aria-label={ariaLabel} />
    </div>
  );
};

export default KeySwitch;
