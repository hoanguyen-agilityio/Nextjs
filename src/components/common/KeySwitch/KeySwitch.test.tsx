import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import KeySwitch from '.';

describe('KeySwitch component', () => {
  test('renders with default type', () => {
    const { container } = render(
      <KeySwitch
        title="Generate License Keys"
        ariaLabel="Generate License Keys"
        describe="Issue each customer a unique license key after purchase"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
