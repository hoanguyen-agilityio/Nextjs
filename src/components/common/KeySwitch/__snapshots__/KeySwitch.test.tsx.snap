// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeySwitch component renders with default type 1`] = `
<div>
  <div
    class="flex justify-between undefined"
  >
    <div>
      <span
        class="text-lg text-black-900 dark:text-white"
      >
        Generate License Keys
      </span>
      <p
        class="font-normal text-xsm text-ratio-200 dark:text-grayBlue-400"
      >
        Issue each customer a unique license key after purchase
      </p>
    </div>
    <label
      aria-label="Generate License Keys"
      class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent select-none"
    >
      <span
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          aria-labelledby=":r0:"
          role="switch"
          type="checkbox"
          value=""
        />
      </span>
      <span
        aria-hidden="true"
        class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:bg-primary group-data-[selected=true]:text-primary-foreground w-12 h-7 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background"
      >
        <span
          class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-5 h-5 text-small group-data-[selected=true]:ml-5 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-5 transition-all group-data-[pressed=true]:w-6 group-data-[selected]:group-data-[pressed]:ml-4"
        />
      </span>
    </label>
  </div>
</div>
`;
